os/kernel/pthread/pthread_keydelete.c

```diff
@@ -111,13 +79,36 @@ int pthread_key_delete(pthread_key_t key)
 {
 	struct pthread_tcb_s *rtcb = (FAR struct pthread_tcb_s *)this_task();
 	struct task_group_s *group = rtcb->cmn.group;
+	struct pthread_key_s *cur_key;
 
 	DEBUGASSERT(group && (rtcb->cmn.flags & TCB_FLAG_TTYPE_MASK) == TCB_FLAG_TTYPE_PTHREAD);
 
-	if (key < PTHREAD_KEYS_MAX && group->tg_keys[key] == IN_USE) {
-		rtcb->pthread_data[key].data = NULL;
-		rtcb->pthread_data[key].destructor = NULL;
-		group->tg_keys[key] = NOT_IN_USE;
+	/* Check key validation */
+
+	if (key < PTHREAD_KEYS_MAX) {
```

Comment:
At some places,  key validation is done as
if (group->tg_nkeys == PTHREAD_KEYS_MAX) 

here key validation is done as  if (key < PTHREAD_KEYS_MAX) {

Though logic is correct,  I think it's better to mention in comments that, keys are starting from 0 and go until ( PTHREAD_KEYS_MAX - 1  ) or else it's confusing as same variable (PTHREAD_KEYS_MAX) used to validate 2 conditions